### ISSUE TITLE
Uncaught `concurrent.futures._base.InvalidStateError: CANCELLED`: job status remains at "running"

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal-server#482]
I've been trying to rerun the 23 well example again on the newest Fractal release (see https://github.com/fractal-analytics-platform/fractal-demos/tree/main/examples/user_specific_examples/01_CardiacDifferentiation_23wells). I exceeded memory in the Yokogawa to Zarr task according to the slurm.err logs. But the status of the job in Fractal according to `fractal job show` remains "running". And the last entry in the workflow.log is:
`2023-01-30 15:29:46,621; INFO; SUBMIT 1-th task (name="Convert Yokogawa to OME-Zarr", executor=cpu-low).`
(neither finished nor failed).

But nothing is running on slurm anymore (and hasn't been for half a day).

I get the following message in the slurm.err logs:
`slurmstepd: Exceeded job memory limit at some point.`

Weirdly though, the main job logs contain this in many wells:
`2023-01-30 17:25:57,557; INFO; END yokogawa_to_ome_zarr task`

----

Looking at the server, I see the following uncaught error:
```
Exception in thread Thread-135:
Traceback (most recent call last):
  File "/data/homes/fractal/.conda/envs/fractal-server-1.0.2/lib/python3.8/site-packages/fractal_server/app/runner/_slurm/executor.py", line 483, in _completion
    fut.set_exception(job_exc)
  File "/data/homes/fractal/.conda/envs/fractal-server-1.0.2/lib/python3.8/concurrent/futures/_base.py", line 547, in set_exception
    raise InvalidStateError('{}: {!r}'.format(self._state, self))
concurrent.futures._base.InvalidStateError: CANCELLED: <Future at 0x7f7c91fc28e0 state=cancelled>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/data/homes/fractal/.conda/envs/fractal-server-1.0.2/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/data/homes/fractal/.conda/envs/fractal-server-1.0.2/lib/python3.8/site-packages/cfut/__init__.py", line 65, in run
    self.check(i)
  File "/data/homes/fractal/.conda/envs/fractal-server-1.0.2/lib/python3.8/site-packages/cfut/__init__.py", line 195, in check
    self.callback(finished_id)
  File "/data/homes/fractal/.conda/envs/fractal-server-1.0.2/lib/python3.8/site-packages/fractal_server/app/runner/_slurm/executor.py", line 488, in _completion
    fut.set_exception(e)
  File "/data/homes/fractal/.conda/envs/fractal-server-1.0.2/lib/python3.8/concurrent/futures/_base.py", line 547, in set_exception
    raise InvalidStateError('{}: {!r}'.format(self._state, self))
concurrent.futures._base.InvalidStateError: CANCELLED: <Future at 0x7f7c91fc28e0 state=cancelled>
```

I'm not sure what cancelled the jobs. Maybe the out of memory? Maybe something went wrong on the cluster overnight? Maybe someone cancelled my jobs (probably not, but who knows).

Will retry with more memory, let's see. But we should catch this exception as well

**Comment by jluethi** [comment]
All running on fractal-server 1.0.2, checked with fractal-client 1.0.1

**Comment by tcompa** [comment]
This should be fixed by #497, available in fractal-server=1.0.8.
If we can try to reproduce the error, let's verify that it is handled correctly.
